src/buildHTML.js

```diff
@@ -88,146 +64,173 @@ export const buildExpression = function(
     surrounding: [?DomType, ?DomType] = [null, null],
 ): HtmlDomNode[] {
     // Parse expressions into `groups`.
-    const rawGroups: HtmlDomNode[] = [];
+    const groups: HtmlDomNode[] = [];
     for (let i = 0; i < expression.length; i++) {
         const output = buildGroup(expression[i], options);
         if (output instanceof DocumentFragment) {
             const children: HtmlDomNode[] = output.children;
-            rawGroups.push(...children);
+            groups.push(...children);
         } else {
-            rawGroups.push(output);
+            groups.push(output);
+        }
+    }
+
+    // If `expression` is a partial group, let the parent handle spacings
+    // to avoid processing groups multiple times.
+    if (!isRealGroup) {
+        return groups;
+    }
+
+    let glueOptions = options;
+    if (expression.length === 1) {
+        const node = checkNodeType(expression[0], "sizing") ||
+            checkNodeType(expression[0], "styling");
+        if (!node) {
```

Comment:
Not being able define variables inside `if` blocks is a drag.  It's too bad we can't do:
```
if (const node = checkNodeType(expression[0], "sizing")) {
   ...
} else if (const node = checkNodeType(expression[0], "styling")) {
   ...
}
```
oh well.